internal/resources/providers/awslib/config.go

```diff
@@ -18,28 +18,102 @@
 package awslib
 
 import (
+	"context"
+	"fmt"
 	"net/http"
+	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
 	libbeataws "github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
+
+	"github.com/elastic/cloudbeat/internal/config"
 )
 
+func awsConfigRetrier() aws.Retryer {
+	return retry.NewStandard(func(o *retry.StandardOptions) {
+		o.Retryables = append(o.Retryables, retry.RetryableHTTPStatusCode{
+			Codes: map[int]struct{}{
+				http.StatusTooManyRequests: {},
+			},
+		})
+	})
+}
+
 func InitializeAWSConfig(cfg libbeataws.ConfigAWS) (*aws.Config, error) {
 	awsConfig, err := libbeataws.InitializeAWSConfig(cfg)
 	if err != nil {
 		return nil, err
 	}
 
-	awsConfig.Retryer = func() aws.Retryer {
-		return retry.NewStandard(func(o *retry.StandardOptions) {
-			o.Retryables = append(o.Retryables, retry.RetryableHTTPStatusCode{
-				Codes: map[int]struct{}{
-					http.StatusTooManyRequests: {},
-				},
-			})
-		})
-	}
+	awsConfig.Retryer = awsConfigRetrier
 
 	return &awsConfig, nil
 }
+
+func CloudConnectorsExternalID(resourceID, externalIDPart string) string {
+	return fmt.Sprintf("%s-%s", resourceID, externalIDPart)
+}
+
+func InitializeAWSConfigCloudConnectors(ctx context.Context, cfg config.AwsConfig) (*aws.Config, error) {
```

Comment:
I wonder if we better to have a more generic flow in case we need longer chain. 
The chain can be part of the agent policy and be used also for non cloud connector use cases

Reply:
I thought about implementing it in a "loop" mode, but: 

1. this will eventually be moved to libbeat so a more generic approach can be implemented there
2. After some tests that I just completed, there are even fewer steps now; it's the init and 2 extras assume instead of 3. 

I believe a more generic approach (if it is worth the effort) can be implemented when that code gets transferred to libbeat. 

Wdyt?

Reply:
Implemented it in a loop-like logic.